prow/plugins/plugins.go

```diff
@@ -131,14 +131,32 @@ type PluginAgent struct {
 // target for plugin Configuration
 type Configuration struct {
 	// Repo (eg "k/k") -> list of handler names.
-	Plugins         map[string][]string `json:"plugins,omitempty"`
-	Triggers        []Trigger           `json:"triggers,omitempty"`
-	Heart           Heart               `json:"heart,omitempty"`
-	MilestoneStatus MilestoneStatus     `json:"milestonestatus,omitempty"`
-	Slack           Slack               `json:"slack,omitempty"`
-	// ConfigUpdater holds config for the config-updater plugin.
-	ConfigUpdater ConfigUpdater `json:"config_updater,omitempty"`
-	Blockades     []Blockade    `json:"blockades,omitempty"`
+	Plugins map[string][]string `json:"plugins,omitempty"`
+
+	// ExternalPlugins is a list of external plugin names mapped
+	// to Kubernetes Service names.
+	ExternalPlugins []ExternalPlugin `json:"external_plugins"`
+
+	// Prow build-in plugins
+	Triggers        []Trigger       `json:"triggers,omitempty"`
+	Heart           Heart           `json:"heart,omitempty"`
+	MilestoneStatus MilestoneStatus `json:"milestonestatus,omitempty"`
+	Slack           Slack           `json:"slack,omitempty"`
+	ConfigUpdater   ConfigUpdater   `json:"config_updater,omitempty"`
+	Blockades       []Blockade      `json:"blockades,omitempty"`
+}
+
+// ExternalPlugin holds configuration for registering an external
+// plugin in prow.
+type ExternalPlugin struct {
+	// Name of the plugin.
+	Name string `json:"name"`
+	// Service is the name of the Kubernetes Service that exposes
+	// the plugin.
+	Service string `json:"service"`
+	// Events are the events that need to be demuxed by the hook
+	// server to the external plugin.
+	Events []string `json:"events"`
```

Comment:
Can we get an enum for these? Might be nice to validate them on startup

Reply:
One of my first thoughts but I preferred to keep it simple for start.